### ISSUE TITLE
emit a warning on a component/group page if a custom processor cannot process it

### ISSUE THREAD

**Issue by djamaile** [backstage/backstage#9138]
## Feature Suggestion

So, when you have a custom processor and it is encountering an error and cannot process it further you want to display this on the component it self or on the owner page. It is currently possible to emit such an error/warning on the component page but only if the component was already registered. If it wasn't registered before then it is not possible to show this to the owner of the component. 

Because you cannot show this on the component if it isn't registered, you'd like to emit this warning on the groups own page. But I don't think this is currently possible because the `emit` function takes in a `location` parameter and every group has the same location. So, I think you would be essentially showing this error on every groups page (at least for LDAP). 

## Possible Implementation

Not sure how an implementation could look like. But in my discussion with @benjdlambert this were hist thoughts:

'Just read through the original message again, I wonder if there is a use case for creating a shell component for things that fail processing for custom processors. Like we can be pretty sure that our processors that we provide should succeed otherwise things will break, but custom provided stuff maybe doesn't need to happen if we can still build some form of entity out of it, and be able to display those errors on an entity page
...
There's a question about wether this should be in the catalog processors or not, but I think we can talk about that in the ticket, I don't think right now that this warrants some form of checking service, and it could be that the processor could first warn, and then hard fail after a few weeks or something.'

## Context

Basically when a custom processor encounters an error I want to notify the owner of that Component. Either by displaying a warning on the group or the components page.

**Comment by Rugvip** [comment]
Yep, definitely need this!

I could've sworn we already had an issue laying around that tracked this, but can't find it 😅

**Comment by djamaile** [comment]
@Rugvip nice! Is this something the maintainers want to do themselves?

**Comment by jhaals** [comment]
Totally agree we need/want this but it requires some exploration and finally a mapping on how and where different errors should be surfaced almost in a case by case bases to be able to come up with a good implementation, especially with new and unexperienced users in mind.

The tricky bit is how to surface some of these errors. Discovery processors for example where you automatically scan your org for new catalog-info files. Let's say one of the files are malformed and the entity does not show up, where should the user expect to find the error?

**Comment by djamaile** [comment]
@jhaals Ideally you have some sort of error/tasklist/notification hub which displays the error you and/or parent ancestries have. This would be then displayed in the frontend for example, on the homepage. Obviously this is way bigger then what this issue describes. I guess what I am describing is more related to #639 , but maybe we can start small?

**Comment by jhaals** [comment]
It's what we're thinking at a high level and it can be small/simple as a start but it's important that the trickier use cases that might require rearchitecting certain parts are taken into account.

I'm interpreting #639 more as a way of displaying arbitrary events from other systems in Backstage since kubernetes events are mentioned in the example. This is trickier as many of these "failed to process" cases might not have a user/group that the error can be tied to. For example a broken location does not have any of it. In that case it might make sense to somehow be able to find it by the URL. Discovery is perhaps the one to tackle first since the many of the others are simpler. The errors or processors probably need some identifier that make them possible to query for the user.

**Comment by williamscs** [comment]
@jhaals @Rugvip , would it be possible to re-open this? I know the linked issue kind of handles a more general case, but this would be a good follow-on. Or as OP stated, could this could potentially start small before 639 is completed.
